app/dockerdwrapper.c

```diff
@@ -405,32 +427,44 @@ start_dockerd(const struct settings *settings)
 
   g_strlcpy(msg, "Starting dockerd", msg_len);
 
-  if (use_tls) {
-    const char *ca_path = "/usr/local/packages/dockerdwrapper/ca.pem";
-    const char *cert_path =
-        "/usr/local/packages/dockerdwrapper/server-cert.pem";
-    const char *key_path = "/usr/local/packages/dockerdwrapper/server-key.pem";
+  if (!use_ipc_socket && !use_tcp_socket) {
+    syslog(LOG_ERR,
+           "At least one of IPC socket or TCP socket must be set to \"yes\". "
+           "dockerd will not be started.");
+    goto end;
+  }
 
-    args_offset += g_snprintf(args + args_offset,
-                              args_len - args_offset,
-                              " %s %s %s %s %s %s %s %s",
-                              "-H tcp://0.0.0.0:2376",
-                              "--tlsverify",
-                              "--tlscacert",
-                              ca_path,
-                              "--tlscert",
-                              cert_path,
-                              "--tlskey",
-                              key_path);
-
-    g_strlcat(msg, " in TLS mode", msg_len);
+  if (use_tcp_socket) {
+    if (use_tls) {
```

Comment:
suggest the following change for lines 438-469
```suggestion
    g_strlcat(msg, " with TCP socket", msg_len);
    const uint port = use_tls ? 2376 : 2375;
    args_offset += g_snprintf(args + args_offset,
                              args_len - args_offset,
                              " -H tcp://0.0.0.0:%d",
                              port);
    if (use_tls) {
      const char *ca_path = "/usr/local/packages/dockerdwrapper/ca.pem";
      const char *cert_path =
          "/usr/local/packages/dockerdwrapper/server-cert.pem";
      const char *key_path =
          "/usr/local/packages/dockerdwrapper/server-key.pem";

      args_offset += g_snprintf(args + args_offset,
                                args_len - args_offset,
                                " %s %s %s %s %s %s %s",
                                "--tlsverify",
                                "--tlscacert",
                                ca_path,
                                "--tlscert",
                                cert_path,
                                "--tlskey",
                                key_path);
      g_strlcat(msg, " in TLS mode", msg_len);
    } else {
      args_offset += g_snprintf(
          args + args_offset, args_len - args_offset, " --tls=false");

      g_strlcat(msg, " in unsecured mode", msg_len);
    }

  } else {
    g_strlcat(msg, " without TCP socket.", msg_len);
```

Reply:
I have added the suggested code with minor changes since it was missing to inclide the log message  'in unsecured mode' when TCP not selected